lib/makara/cache.rb

```diff
@@ -20,7 +20,7 @@ def read(key)
       end
 
       def write(key, value, ttl)
-        store.try(:write, key, value, :expires_in => ttl.to_i)
+        store.try(:write, key, value, :expires_in => ttl.to_f)
```

Comment:
Workaround: Store the stickiness duration as the cache value instead of implying it as the TTL. For cache TTL, use `ttl.ceil` so it's at least 1 second.